frontend/gateway/grpcclient/client.go

```diff
@@ -483,8 +484,20 @@ func (c *grpcClient) ResolveImageConfig(ctx context.Context, ref string, opt llb
 }
 
 func (c *grpcClient) BuildOpts() client.BuildOpts {
+	opts := c.opts
+	if c.caps.Supports(pb.CapAttestations) == nil {
```

Comment:
`Supports() == nil` means that capability IS supported.

I think the better behavior in this cases would be to make sure `AddAttestation` is a noop if daemon has no support.